src/test/java/com/sprint/mission/discodeit/service/UserServiceTest.java

```diff
@@ -0,0 +1,165 @@
+package com.sprint.mission.discodeit.service;
+
+import com.sprint.mission.discodeit.dto.data.UserDto;
+import com.sprint.mission.discodeit.dto.request.BinaryContentCreateRequest;
+import com.sprint.mission.discodeit.dto.request.UserCreateRequest;
+import com.sprint.mission.discodeit.dto.request.UserUpdateRequest;
+import com.sprint.mission.discodeit.entity.BinaryContent;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.exception.ErrorCode;
+import com.sprint.mission.discodeit.exception.user.UserAlreadyExistsException;
+import com.sprint.mission.discodeit.exception.user.UserNotFoundException;
+import com.sprint.mission.discodeit.mapper.BinaryContentMapper;
+import com.sprint.mission.discodeit.mapper.UserMapper;
+import com.sprint.mission.discodeit.mapper.UserMapperImpl;
+import com.sprint.mission.discodeit.repository.BinaryContentRepository;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import com.sprint.mission.discodeit.service.basic.BasicUserService;
+import com.sprint.mission.discodeit.storage.BinaryContentStorage;
+import com.sprint.mission.discodeit.support.UserFixture;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mapstruct.factory.Mappers;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTest {
+    private final String USER_EMAIL = "jionu102@naver.com";
+    private final String USERNAME = "jionu102";
+    private final String USER_PASSWORD = "123412341234";
+    private final UUID USER_ID = UUID.randomUUID();
+
+    private final String PROFILE_FILE_NAME = "rex.jpg";
+    private final Long PROFILE_FILE_SIZE = 1024L;
+    private final String PROFILE_CONTENT_TYPE = "image/jpeg";
+    private final byte[] PROFILE_FILE_CONTENT = new byte[1024];
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private BinaryContentRepository binaryContentRepository;
+    @Mock
+    private BinaryContentStorage binaryContentStorage;
+    @Spy
+    private UserMapper userMapper = new UserMapperImpl(){
+        final BinaryContentMapper binaryContentMapper = Mappers.getMapper(BinaryContentMapper.class);
+        @Override
+        public UserDto toDto(User user) {
+
+            return new UserDto(
+                    user.getId(),
+                    user.getUsername(),
+                    user.getEmail(),
+                    binaryContentMapper.toDto(user.getProfile()),
+                    user.getStatus() != null ? user.getStatus().isOnline() : null
+            );
+        }
+    };
```

Comment:
Mapper는 직접 익명클래스로 구현체를 만들기보다는 `Mappers.getMapper()` 메서드 사용하면 가독성이 더 좋아집니다.
UserMapper 내부에 의존하고 있는 BinaryContentMapper은 ReflectionUtil로 주입해주면 됩니다.
```suggestion
    @Spy
    private UserMapper userMapper = Mappers.getMapper(UserMapper.class)
```